SYCL/ESIMD/regression/dgetrf.cpp

```diff
@@ -0,0 +1,479 @@
+      dgetrfnp_left_step<32, 8, 8>(a, lda, info);
+      dgetrfnp_left_step<32, 8, 16>(a, lda, info);
+      dgetrfnp_left_step<32, 8, 24>(a, lda, info);
+    }
+  } else if (m == 64) {
+    if (n == 6)
+      dgetrfnp_left_step<64, 6, 0>(a, lda, info);
+    else if (n == 16) {
+      dgetrfnp_left_step<64, 6, 0>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 6>(a, lda, info);
+      dgetrfnp_left_step<64, 4, 12>(a, lda, info);
+    } else if (n == 32) {
+      dgetrfnp_left_step<64, 6, 0>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 6>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 12>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 18>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 24>(a, lda, info);
+      dgetrfnp_left_step<64, 2, 30>(a, lda, info);
+    } else if (n == 64) {
+      dgetrfnp_left_step<64, 6, 0>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 6>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 12>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 18>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 24>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 30>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 36>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 42>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 48>(a, lda, info);
+      dgetrfnp_left_step<64, 6, 54>(a, lda, info);
+      dgetrfnp_left_step<64, 4, 60>(a, lda, info);
+    }
+  }
+#endif // defined(USE_REF)
+}
+
+void dgetrfnp_batch_strided_c(int64_t m, int64_t n, double *a, int64_t lda,
+                              int64_t stride_a, int64_t *ipiv,
+                              int64_t stride_ipiv, int64_t batch,
+                              int64_t *info) {
+  queue queue((gpu_selector()));
+  auto device = queue.get_device();
+  auto context = queue.get_context();
+  int status;
+
+  CHECK(status = device.is_gpu(), !status);
+
+  double *a_gpu;
+  int64_t *ipiv_gpu;
+  int64_t *info_gpu;
+  CHECK(a_gpu = static_cast<double *>(
+            malloc_shared(stride_a * batch * sizeof(double), device, context)),
+        !a_gpu);
+  CHECK(ipiv_gpu = static_cast<int64_t *>(malloc_shared(
+            stride_ipiv * batch * sizeof(int64_t), device, context)),
+        !ipiv_gpu);
+  CHECK(info_gpu = static_cast<int64_t *>(
+            malloc_shared(batch * sizeof(int64_t), device, context)),
+        !info_gpu);
+
+  memcpy(a_gpu, a, stride_a * batch * sizeof(double));
+
+  sycl::nd_range<1> range(sycl::range<1>{static_cast<size_t>(batch)},
+                          sycl::range<1>{1});
+  try {
+    auto event = queue.submit([&](handler &cgh) {
+      cgh.parallel_for<class dgetrfnp_batch_strided>(
+          range, [=](nd_item<1> id) SYCL_ESIMD_KERNEL {
+            int i = id.get_global_id(0);
+            dgetrfnp_esimd(m, n, &a_gpu[i * stride_a], lda,
+                           &ipiv_gpu[i * stride_ipiv], &info_gpu[i]);
+          });
+    });
+    event.wait();
+  } catch (const sycl::exception &e) {
+    std::cout << "*** EXCEPTION caught: " << e.what() << "\n";
```

Comment:
I think there is a memory leak here. Calls to `free` should be added.

Reply:
Done.